exonum/src/runtime/mod.rs

```diff
@@ -255,9 +258,23 @@ pub trait Runtime: Send + Debug + 'static {
         tx_sender: &ApiSender,
     );
 
-    fn services_api(&self) -> Vec<(String, ServiceApiBuilder)> {
+    /// Collect the full list of API handlers from the runtime for the built-in Exonum API server.
+    ///
+    /// This method is called during an API server restart. Use this method if you do not plan to
+    /// use your own API processing mechanism.
+    ///
+    /// Warning! It is a temporary method which not to break the existing `RustRuntime` code,
+    /// in future it will be removed.
+    #[doc(hidden)]
+    fn api_endpoints(&self, _context: &ApiContext) -> Vec<(String, ServiceApiBuilder)> {
         Vec::new()
     }
+
+    /// Notify runtime about changes in list of service instances.
+    ///
+    /// The purpose of this method is provide building blocks to create your own API processing
+    /// mechanisms.
+    fn notify_api_changes(&self, _context: &ApiContext, _changes: &[ApiChange]) {}
```

Comment:
Is it called before or after the service has been configured? I.e. will its API become externally available to requests before configuration?

Reply:
This method is called last even after `after_commit`.